site/app/libraries/Core.php

```diff
@@ -265,6 +265,56 @@ public function getCourseQueries(): array {
         return $queries;
     }
 
+    public function hasDBPerformanceWarning(): bool {
+        if (count($this->getSubmittyQueries()) + count($this->getCourseQueries()) > 20) {
+            return true;
+        }
+
+        if (($this->course_db !== null && $this->course_db->hasDuplicateQueries()) || ($this->submitty_db !== null && $this->submitty_db->hasDuplicateQueries())) {
+            return true;
+        }
+
+        $queries = [];
+        if ($this->course_debug_stack !== null) {
+            foreach ($this->course_debug_stack->queries as $query) {
+                $queries[] = $query['sql'];
+            }
+        }
+        if ($this->submitty_debug_stack !== null) {
+            foreach ($this->submitty_debug_stack->queries as $query) {
+                $queries[] = $query['sql'];
+            }
+        }
+
+        return count($queries) !== count(array_unique($queries));
+    }
+
+    protected function logPerformanceWarning(): void {
```

Comment:
```suggestion
    private function logPerformanceWarning(): void {
```
Can be private I'm assuming cause there is no child of Core